index.d.ts

```diff
@@ -22,19 +33,52 @@ declare interface TwemojiOptions {
   /**
    * The function to invoke in order to generate image src(s).
    */
-  callback?(icon: string, options: TwemojiOptions): void;
+  callback?: ParseCallback
   /**
+   * The function to invoke in order to generate additional, custom attributes for the image tag.
    * Default () => ({})
+   * @param icon the lower case HEX code point i.e. "1f4a9"
+   * @param variant variant the optional \uFE0F ("as image") variant, in case this info is anyhow meaningful. By default this is ignored.
+   * 
    */
-  attributes?(): void;
+  attributes?(icon: string, variant: string): object;
 }
 
 declare type Twemoji = {
   convert: {
+    /**
+     * Given an HEX codepoint, returns UTF16 surrogate pairs.
+     *
+     * @param   codepoint string  generic codepoint, i.e. '1F4A9'
+     * @return  string  codepoint transformed into utf16 surrogates pair,
+     *          i.e. \uD83D\uDCA9
+     *
+     * @example
+     *  twemoji.convert.fromCodePoint('1f1e8');
+     *  // "\ud83c\udde8"
+     *
+     *  '1f1e8-1f1f3'.split('-').map(twemoji.convert.fromCodePoint).join('')
+     *  // "\ud83c\udde8\ud83c\uddf3"
+     */
     fromCodePoint(hexCodePoint: string): string;
-    toCodePoint(utf16surrogatePairs: string): string;
+
+    /**
+     * Given UTF16 surrogate pairs, returns the equivalent HEX codepoint.
+     *
+     * @param   utf16surrogatePairs   string  generic utf16 surrogates pair, i.e. \uD83D\uDCA9
+     * @param   sep                 string  optional separator for double code points, default='-'
+     * @return  string  utf16 transformed into codepoint, i.e. '1F4A9'
```

Comment:
in JSDoc, types must be wrapped in curly braces (and for `@param`, the type should come before the name)